fetch/client/src/main/scala/endpoints4s/fetch/Endpoints.scala

```diff
@@ -281,68 +282,75 @@ trait EndpointsWithCustomErrors
       request: Request[A],
       response: Response[B],
       a: A
-  )(
-      onload: Either[Throwable, B] => Unit,
-      onerror: Throwable => Unit
-  ): Unit = {
+  ): (js.Promise[B], js.Function1[Unit, Unit]) = {
```

Comment:
Should it be?

```suggestion
  ): (js.Thenable[B], js.Function1[Unit, Unit]) = {
```

Maybe it doesn’t matter, I am not familiar with JS promises.

Reply:
Also, maybe a `Function0[Unit]` would work better?

Reply:
Last, I am not sure whether this is important or not, but we previously created either a `Thenable` or a `Future` depending on the concrete xhr interpreter. Now, we always create a `Thenable`, even when we return a `Future`.